### ISSUE TITLE
Handle scanning responses from sleep proxies

### ISSUE THREAD

**Issue by postlund** [postlund/pyatv#724]
**Short feature/function description**
Continuation of #723. With this story I plan to perform unicast scanning on responses from sleep proxies in order to look up the remaining metadata. I believe this is the last piece to make scanning really robust.

**What needs to be done?**
Whenever a device is sleeping and a sleep proxy returns the zeroconf response, it will only contain PTRs to the services. The remaining data can be retrieved with unicast scanning using the full qualified name. So instead of just `_mediaremotetv._tcp.local` it's possible to use `Kitchen._mediaremotetv._tcp.local` and send that to the sleep proxy. This is needed since multiple devices can be sleeping using the same sleep proxy.

**Is this a breaking change?**
No

**Anything else worth knowing?**
I hope this will make scanning even more robust.

**Comment by postlund** [comment]
Unfortunately there seems to be some annoying gap-times when a device wakes up or goes to sleep. Some times the sleep proxy announces existence of the device (I.e. just PTR records) but does not respond to unicast scans for said device. Some times none of the devices respond to multicast at all, so it is simply not found. I'm gonna try to figure out if there are some potential timeout values that are good. I would prefer to have a scan method that is robust and finds all devices, taking its time doing so, than scanning that is not robust.

**Comment by postlund** [comment]
This is not fun:
```
2020-06-23 12:24:23 DEBUG: Received DNS response ...
2020-06-23 12:24:51 DEBUG: Received DNS response ...
```
The code multicast for services every second and at some point no response is received (either from my ATV3 or sleep proxy) for **28 seconds**. That is a looong time. Why is that...

**Comment by postlund** [comment]
This is even worse:
```
2020-06-23 12:31:21 DEBUG: Received DNS response
2020-06-23 12:32:01 DEBUG: Received DNS response
```

**Comment by postlund** [comment]
Ok, so I'm stupid. Disregard my previous comments as I *only* looked at responses from the device and not the sleep proxy. Will do a new analysis.

**Comment by postlund** [comment]
So, I did a 20min continuous scan (new multicast packet every second) and this is what I found:
* When the device is awake, it responds to every request. The same goes for when it is sleeping, the sleep proxy responds to every request instead. One request might be missed every now and then, which isn't a big deal since I re-send the request multiple times.
* When the device is about to go to sleep, the device and the sleep proxy will announce the services simultaneously (with no gaps when the device goes to sleep) for about ten seconds (.1 sleep proxy, .103 ATV):
```
13:34:33 10.0.10.103
13:34:34 10.0.10.103
13:34:35 10.0.10.1
13:34:35 10.0.10.103
13:34:36 10.0.10.1
13:34:36 10.0.10.103
13:34:37 10.0.10.1
13:34:37 10.0.10.103
13:34:38 10.0.10.1
13:34:38 10.0.10.103
13:34:39 10.0.10.1
13:34:39 10.0.10.103
13:34:40 10.0.10.1
13:34:40 10.0.10.103
13:34:41 10.0.10.1
13:34:41 10.0.10.103
13:34:42 10.0.10.1
13:34:42 10.0.10.103
13:34:43 10.0.10.1
13:34:43 10.0.10.103
13:34:44 10.0.10.1
13:34:44 10.0.10.103
13:34:45 10.0.10.1
13:34:46 10.0.10.1
13:34:47 10.0.10.1
```
* When the device wakes up, responses stop coming for roughly ten seconds from any of the devices. I've seen times between 2-10 seconds, usually around 7-8s.
```
13:34:04 10.0.10.1
13:34:05 10.0.10.1
13:34:06 10.0.10.1
13:34:15 10.0.10.103
13:34:16 10.0.10.103
13:34:17 10.0.10.103
```

I also do believe that it is not possible to send unicast requests to the sleep proxy during the "grace period", when the device is about to go to sleep. It will just ignore them and not respond (not confirmed to 100% though).

So, a scanning period would need to be a bit over 10s to likely be considered reliable. At least 10s for multicast but also some time to wait for potential unicast responses. Not sure how I want to handle this yet...
